cmd/juju/action/call.go

```diff
@@ -293,11 +313,69 @@ func (c *callCommand) Run(ctx *cmd.Context) error {
 			return err
 		}
 		d := FormatActionResult(result)
-		d["id"] = tag.Id()       // Action ID is required in case we timed out.
-		d["unit"] = unitTag.Id() // Formatted unit is nice to have.
-		info[result.Action.Receiver] = d
+		d["id"] = tag.Id() // Action ID is required in case we timed out.
+		info[unitTag.Id()] = d
+	}
+
+	if c.out.Name() == "plain" {
+		return c.out.WriteFormatter(ctx, c.printPlainOutput, info)
+	} else {
+		return c.out.Write(ctx, info)
+	}
+}
+
+func (c *callCommand) printPlainOutput(writer io.Writer, value interface{}) error {
+	info, ok := value.(map[string]interface{})
+	if !ok {
+		return errors.New("unexpected value")
```

Comment:
In tabular format, we also state what type the input is. Worthwhile adding it here too I think :)

Reply:
good pickup